server/controllers/member/addMember.js

```diff
@@ -1,3 +1,34 @@
-module.exports = (req, res) => {
-  res.send('signup -- add new member');
+const { sign } = require('jsonwebtoken');
+
+const { addMmeber } = require('../../database/queries/members/addMember');
+const { checkEmail } = require('../../database/queries/authentication/checkEmail');
+const { checkUsername } = require('../../database/queries/authentication/getUser');
+const { singUpSchema } = require('../../helpers/validation-schema');
+const { hashingPass } = require('../../helpers/hashPassword');
+
+module.exports = (req, res, next) => {
+  const memberInfo = { ...req.body };
+  singUpSchema.validate(memberInfo)
+    .then(() => {
+      checkUsername(memberInfo.username)
+        .then(({ rows: validUsername }) => {
+          if (validUsername[0]) throw next({ code: 400, msg: 'The username already exists ' });
+          else return checkEmail(memberInfo.email);
+        })
+        .then(({ rows: validEmail }) => {
+          if (validEmail[0]) throw next({ code: 400, msg: 'The email already exists' });
+          else return hashingPass(memberInfo.pass);
+        })
+        .then(hashedPass => addMmeber({ ...memberInfo, pass: hashedPass }))
+        .then(({ rows: member }) => {
+          const payLoad = {
+            id: member[0].id, username: member[0].username, avatar: member[0].avatar,
+          };
+          const jwt = sign(payLoad, process.env.SECRET);
+          res.cookie('jwt', jwt, { maxAge: 7200000 });
```

Comment:
@Fatmasiam You could refactor these 5 lines (`const payload =, const jwt =, res.cookie`) into a function in another file, similar to how you have functions for checkEmail, checkUsername, and so on.  That would help this file be a little shorter and easier to read - it will then look like a series of steps, but the details of each step wouldn't clutter the file, if that makes sense.

Reply:
thanx for the suggestion :100: